infrastructure/storageprovider/dropbox.go

```diff
@@ -89,11 +107,19 @@ func (d *dropbox) Upload(cred domain.StorageProviderCredential, file io.Reader,
 	}
 
 	// do the request
-	_, err = client.Do(req)
+	res, err := client.Do(req)
 	if err != nil {
 		return err
 	}
 
+	if res.StatusCode != http.StatusOK {
+		byteRes, _ := io.ReadAll(res.Body)
```

Comment:
I think I'll make the `mapToDropboxError` recieve `io.Reader` since the byte array is still needed if the response is a JSON (http code 400 errors returns plaintext).

Reply:
ohh yeah, right. I missed that one.

anyway, help to add `res.Body.Close()` to close the response body. Thanks!

Reply:
Got it